inbm/dispatcher-agent/dispatcher/common/dispatcher_state.py

```diff
@@ -59,29 +67,42 @@ def is_dispatcher_state_file_exists() -> bool:
 def consume_dispatcher_state_file(read: bool = False) -> DispatcherState | None:
     """Read dispatcher state file and return state object, clearing state file on success
 
-    @param read: set to True when only file info needs to be read without removing the file
-    @return: state object from state file; None on error
+    Try old location first, then new location.
+
+    @param read: Set to True to read file info without removing the file
+    @return: State object from state file; None on error
     """
-    logger.debug("")
-    fd = None
+    logger.debug("Starting consume_dispatcher_state_file")
+
     state = None
-    try:
-        logger.debug("attempting to open file " + str(DISPATCHER_STATE_FILE))
-        fd = builtins.open(DISPATCHER_STATE_FILE, 'rb')
-        logger.debug("attempting to unpickle from state file")
-        # override S301 because we are unpickling from trusted file
-        state = pickle.load(fd)  # nosec
-        logger.debug("unpickling succeeded")
-        logger.debug(f"Disp State file info: {state}")
-    except (OSError, pickle.UnpicklingError, AttributeError, FileNotFoundError) as e:
-        logger.exception(f"Exception while extracting dispatcher state from disk: {e}")
-        raise DispatcherException("Exception while extracting dispatcher state from disk") from e
-    finally:
-        if fd:
-            fd.close()
-        if read:
-            return state
-    clear_dispatcher_state()
+    state_files = [OLD_DISPATCHER_STATE_FILE, NEW_DISPATCHER_STATE_FILE]
+
+    for state_file in state_files:
+        try:
+            logger.debug(f"Attempting to open file {state_file}")
+            with builtins.open(state_file, 'rb') as fd:
```

Comment:
Should we check the existence of the state file before open?

Reply:
@yengliong93 here we try both files, and break on the first one that succeeds; if both fail, `state` is set to `None`  -- the exceptions handle cases where the file doesn't exist